### PR TITLE
[QoL] Change Dex filter name to Caught

### PR DESCRIPTION
**Pull request by KimJeongSun**
## What are the changes?

Rename the `Dex` filter to `Caught`.

## Why am I doing these changes?

Before adding filters like HA (Hidden Ability) and Pokerus, we planned to include filters like HA, gender, and Ability within the current Dex filter, hence the name Dex.
However, after several tests, it became apparent that implementing these filters in such a manner was challenging. [related discord message](https://discord.com/channels/1125469663833370665/1267339517371875361/1269680312330358918)

As a result, the HA filter was added to the Misc filter, and the other filters will also likely not be included in the Dex filter in the future.  

To prevent confusion for users using or developing this filter further, we are changing the name from Dex to Caught. This ensures that the current functionality of the Dex filter does not overlap with the functionalities of other filters.

## What did change?

Renamed the `Dex` filter to `Caught`. Consequently, the locale entries and the corresponding code variables have been updated to reflect this change.

### Screenshots/Videos

image

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [ ] Have I considered writing automated tests for the issue?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

File: src/locales/zh_TW/filter-bar.ts [hunk 1]
```diff
@@ -4,7 +4,7 @@ export const filterBar: SimpleTranslationEntries = {
   "genFilter": "世代",
   "typeFilter": "屬性",
   "dexFilter": "Dex",
```
Comment (MokaStitcher) [inline src/locales/zh_TW/filter-bar.ts:6]: changed the wrong entry here

**Review by MokaStitcher — CHANGES_REQUESTED** [review]

File: src/locales/zh_TW/filter-bar.ts (hunk 1 above)
Comment (KimJeongSun) [reply src/locales/zh_TW/filter-bar.ts:6]: you saved me

**Review by MokaStitcher — APPROVED** [review]
looks good

File: src/locales/fr/filter-bar.ts [hunk 2]
```diff
@@ -3,7 +3,7 @@ import { SimpleTranslationEntries } from "#app/interfaces/locales";
 export const filterBar: SimpleTranslationEntries = {
   "genFilter": "Gen",
   "typeFilter": "Type",
-  "dexFilter": "Dex",
+  "caughtFilter": "Caught",
```
Comment (Adri1) [inline src/locales/fr/filter-bar.ts:6]:
```suggestion
  "caughtFilter": "Capturés",
```

File: src/locales/pt_BR/filter-bar.ts [hunk 3]
```diff
@@ -3,7 +3,7 @@ import { SimpleTranslationEntries } from "#app/interfaces/locales";
 export const filterBar: SimpleTranslationEntries = {
   "genFilter": "Ger.",
   "typeFilter": "Tipo",
-  "dexFilter": "Dex",
+  "caughtFilter": "Caught",
```
Comment (josericardo-fo) [inline src/locales/pt_BR/filter-bar.ts:6]:
```suggestion
  "caughtFilter": "Capturado",
```

File: src/locales/ko/filter-bar.ts [hunk 4]
```diff
@@ -3,7 +3,7 @@ import { SimpleTranslationEntries } from "#app/interfaces/locales";
 export const filterBar: SimpleTranslationEntries = {
   "genFilter": "세대",
   "typeFilter": "타입",
-  "dexFilter": "도감",
+  "caughtFilter": "Caught",
```
Comment (EnochG1) [inline src/locales/ko/filter-bar.ts:6]:
```suggestion
  "caughtFilter": "포획",
```

File: src/locales/zh_CN/filter-bar.ts [hunk 5]
```diff
@@ -3,7 +3,7 @@ import { SimpleTranslationEntries } from "#app/interfaces/locales";
 export const filterBar: SimpleTranslationEntries = {
   "genFilter": "世代",
   "typeFilter": "属性",
-  "dexFilter": "Dex",
+  "caughtFilter": "Caught",
```
Comment (40chyan) [inline src/locales/zh_CN/filter-bar.ts:6]:
```suggestion
  "caughtFilter": "捕获",
```

File: src/locales/zh_TW/filter-bar.ts [hunk 6]
```diff
@@ -3,7 +3,7 @@ import { SimpleTranslationEntries } from "#app/interfaces/locales";
 export const filterBar: SimpleTranslationEntries = {
   "genFilter": "世代",
   "typeFilter": "屬性",
-  "dexFilter": "Dex",
+  "caughtFilter": "Caught",
```
Comment (40chyan) [inline src/locales/zh_TW/filter-bar.ts:6]:
```suggestion
  "caughtFilter": "捕獲",
```

**Review by 40chyan — COMMENTED** [review]
zh

**Pull request merged**